File: web/src/app/webapi/models/candidates.ts
/* tslint:disable */
import { Organizations } from './organizations';
import { Qualifications } from './qualifications';
import { CandidateDocuments } from './candidate-documents';

/**
 */
export class Candidates {
    isReadyForInterview?: boolean;
    candidateId?: string;
    firstName: string;
    lastName?: string;
    experienceYear: number;
    experienceMonth: number;
    qualificationId: number;
    organizationId: number;
    organisation?: Organizations;
    description?: string;
    isApproved?: boolean;
    gender: number;
    qualification?: Qualifications;
    candidateDocuments?: CandidateDocuments[];
    isActive?: boolean;
    createdBy?: string;
    createdDate?: string;
    modifiedBy?: string;
    modifiedDate?: string;
    isDeleted?: boolean;
    deletedBy?: string;
    deletedDate?: string;
}

File: api/RS.API/RS.Entity/Models/Candidates.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace RS.Entity.Models
{
    public partial class Candidates : BaseEntity
    {
        public Candidates()
        {
            CandidateDocuments = new HashSet<CandidateDocuments>();
        }

        [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public Guid CandidateId { get; set; }

        [Required]
        public int Gender { get; set; }

        [Required, MaxLength(50)]
        public string FirstName { get; set; }

        [MaxLength(50)]
        public string LastName { get; set; }

        [Required]
        public int ExperienceYear { get; set; }

        [Required]
        public int ExperienceMonth { get; set; }

        [Required]
        public int QualificationId { get; set; }

        [Required]
        public int OrganizationId { get; set; }

        public Organizations Organisation { get; set; }

        [MaxLength(500)]
        public string Description { get; set; }

        public bool IsApproved { get; set; }

        public bool IsReadyForInterview { get; set; }

        public Qualifications Qualification { get; set; }

        public ICollection<CandidateDocuments> CandidateDocuments { get; set; }
    }
}
